v3/hpc_symmetric3x3.hpp

```diff
@@ -280,8 +283,59 @@ trace(symmetric3x3<T> const x) noexcept {
 
 template <class T>
 HPC_ALWAYS_INLINE HPC_HOST_DEVICE constexpr symmetric3x3<T>
-deviator(symmetric3x3<T> const x) noexcept {
-  return x - ((1.0 / 3.0) * trace(x));
+symmetric_part(matrix3x3<T> const x) noexcept {
+  // The constructor that takes a full matrix implicity takes its symmetric part
+  return symmetric3x3<T>(x);
+}
+
+template <class T>
+HPC_ALWAYS_INLINE HPC_HOST_DEVICE constexpr symmetric3x3<T>
+isotropic_part(symmetric3x3<T> const x) noexcept {
+  return ((1.0 / 3.0) * trace(x)) * symmetric3x3<T>::identity();
+}
+
+template <class T>
+HPC_ALWAYS_INLINE HPC_HOST_DEVICE constexpr symmetric3x3<T>
+deviatoric_part(symmetric3x3<T> const x) noexcept {
```

Comment:
changed name from `deviator` to `deviatoric_part` since, in the mechanics literature, `deviator` more often than not refers to the deviatoric part of the stress tensor.  With the addition of the `hyper_ep` model, we need the deviatoric part of different deformations measures - thus the change of the name to the more general `deviatoric_part`.  I'm find with a revert of the name, just a suggestion :)